### PR TITLE
Don't change the selection if the target element is already selected

### LINKED ISSUE #1430: Sometimes, do not change selection on simulateSelect
In `simulateSelect`, we move the caret to the end.  Probably that is the wrong thing to do if the selection is already in the target element (for example, after find).

### PR DESCRIPTION
**Pull request by mrmr1993**
Fixes #1430.

### CODE REVIEW

File: content_scripts/vimium_frontend.coffee [hunk 1]
```diff
@@ -764,7 +757,7 @@ selectFoundInputElement = ->
   # heuristic of checking that the last anchor node is an ancestor of our element.
   if (findModeQueryHasResults && document.activeElement &&
       DomUtils.isSelectable(document.activeElement) &&
-      isDOMDescendant(findModeAnchorNode, document.activeElement))
+      DomUtils.isDescendant(findModeAnchorNode, document.activeElement))
```
Comment (smblott-github) [inline content_scripts/vimium_frontend.coffee]: Suggest revert name to `isDOMDescendant`, unless there's a good reason to change it (?).

File: lib/dom_utils.coffee [hunk 2]
```diff
@@ -141,11 +141,20 @@ DomUtils =
...
+      @isDescendant element, range.commonAncestorContainer
+    else
+      false
+
   simulateSelect: (element) ->
     element.focus()
-    # When focusing a textbox, put the selection caret at the end of the textbox's contents.
-    # For some HTML5 input types (eg. date) we can't position the caret, so we wrap this with a try.
-    try element.setSelectionRange(element.value.length, element.value.length)
+    unless @isSelected
```
Comment (smblott-github) [inline lib/dom_utils.coffee]: Perhaps:

```coffeescript
@isSelected element
```

Comment (mrmr1993) [inline lib/dom_utils.coffee]: Oops!

**Comment by mrmr1993** [comment]
Updated.

**Comment by smblott-github** [comment]
This doesn't seem to work as expected, @mrmr1993.  Could you check, please?

- Go [here](http://static.smblott.org/inputs.html).
- `/sssw<Enter>`
- `f`, and choose the first/selected text area.

I expect the selection to remain selected.  Instead, the caret is nevertheless moved to the end.

**Comment by smblott-github** [comment]
Closing.  Fixed in #1436.

**Pull request closed without merging**